lib/loglevel.js

```diff
@@ -158,10 +158,12 @@
           if (typeof storedLevel === undefinedType) {
               try {
                   var cookie = window.document.cookie;
-                  var location = cookie.indexOf(
-                      encodeURIComponent(storageKey) + "=");
+                  var cookieName = encodeURIComponent(storageKey);
+                  var location = cookie.indexOf(cookieName + "=");
                   if (location !== -1) {
-                      storedLevel = /^([^;]+)/.exec(cookie.slice(location))[1];
+                      storedLevel = /^([^;]+)/.exec(
+                          cookie.slice(location + cookieName.length + 1)
+                      )[1];
```

Comment:
This is the actual bugfix. The rest is just cleaning up bad test code.

Reply:
> Yep, this makes sense. Any idea why this worked in phantomjs though?

It didn't! One of the reasons I switched to showing the tests as skipped was to verify for myself that *none of the cookie tests run in PhantomJS*. It turns out [`document.cookie` is just always empty and cannot be set](https://github.com/ariya/phantomjs/issues/11222), which causes all the cookie-related and persistence-related tests to be skipped.